plugins/postcss-custom-properties/src/transform-value-ast.ts

```diff
@@ -63,6 +63,7 @@ export default function transformValueAST(root: valueParser.ParsedValue, customP
 
 			if (nodes.length) {
 				parent.nodes.splice(index, 1, ...nodes);
+				parent.nodes.forEach((x) => ancestry.set(x, parent));
```

Comment:
When inserting new nodes the ancestry also needs to be updated to contain the newly inserted nodes.